include/nbl/builtin/glsl/ext/Scan/default_sweep_common.glsl

```diff
@@ -0,0 +1,116 @@
+#ifndef _NBL_GLSL_EXT_SCAN_DEFAULT_SWEEP_COMMON_GLSL_INCLUDED_
+#define _NBL_GLSL_EXT_SCAN_DEFAULT_SWEEP_COMMON_GLSL_INCLUDED_
+
+#ifndef _NBL_GLSL_WORKGROUP_SIZE_
+#define _NBL_GLSL_WORKGROUP_SIZE_ 256
+layout(local_size_x = _NBL_GLSL_WORKGROUP_SIZE_) in;
+#endif
+
+#ifndef _NBL_GLSL_EXT_SCAN_STORAGE_TYPE_
+#error "_NBL_GLSL_EXT_SCAN_STORAGE_TYPE needs to be defined to any of uint/int/float"
+#endif
+
+#define nbl_glsl_ext_Scan_Storage_t _NBL_GLSL_EXT_SCAN_STORAGE_TYPE_
+
+#ifndef _NBL_GLSL_EXT_SCAN_INPUT_SET_DEFINED_
+#define _NBL_GLSL_EXT_SCAN_INPUT_SET_DEFINED_ 0
+#endif
+
+#ifndef _NBL_GLSL_EXT_SCAN_INPUT_BINDING_DEFINED_
+#define _NBL_GLSL_EXT_SCAN_INPUT_BINDING_DEFINED_ 0
+#endif
+
+#ifndef _NBL_GLSL_EXT_SCAN_INPUT_DESCRIPTOR_DEFINED_
+
+layout(set = _NBL_GLSL_EXT_SCAN_INPUT_SET_DEFINED_, binding = _NBL_GLSL_EXT_SCAN_INPUT_BINDING_DEFINED_, std430) buffer InoutBuffer
+{
+	nbl_glsl_ext_Scan_Storage_t inout_values[];
+};
+
+#define _NBL_GLSL_EXT_SCAN_INPUT_DESCRIPTOR_DEFINED_
+#endif
+
+#include "nbl/builtin/glsl/ext/Scan/parameters_struct.glsl"
+#include "nbl/builtin/glsl/ext/Scan/parameters.glsl"
+
+#ifndef _NBL_GLSL_EXT_SCAN_PUSH_CONSTANTS_DEFINED_
+
+layout(push_constant) uniform PushConstants
+{
+	layout(offset = 0) nbl_glsl_ext_Scan_Parameters_t params;
+} pc;
+
+#define _NBL_GLSL_EXT_SCAN_PUSH_CONSTANTS_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_EXT_SCAN_GET_PARAMETERS_DEFINED_
+nbl_glsl_ext_Scan_Parameters_t nbl_glsl_ext_Scan_getParameters()
+{
+	return pc.params;
+}
+#define _NBL_GLSL_EXT_SCAN_GET_PARAMETERS_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_EXT_SCAN_SET_DATA_DEFINED_
+
+void nbl_glsl_ext_Scan_setData(in uint idx, in nbl_glsl_ext_Scan_Storage_t val)
+{
+	if (gl_GlobalInvocationID.x < nbl_glsl_ext_Scan_Parameters_t_getElementCountPass())
+		inout_values[idx] = val;
+}
+
+#define _NBL_GLSL_EXT_SCAN_SET_DATA_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_EXT_SCAN_GET_PADDED_DATA_DEFINED_
+
+nbl_glsl_ext_Scan_Storage_t nbl_glsl_ext_Scan_getPaddedData(in uint idx, in nbl_glsl_ext_Scan_Storage_t pad_val, bool is_upsweep)
+{
+	nbl_glsl_ext_Scan_Storage_t data = pad_val;
+	if (is_upsweep)
+	{
+		if (gl_GlobalInvocationID.x < nbl_glsl_ext_Scan_Parameters_t_getElementCountPass())
+			data = inout_values[idx];
+	}
+	else
+	{
+		if (idx < nbl_glsl_ext_Scan_Parameters_t_getElementCountTotal())
+			data = inout_values[idx];
+	}
+
+	return data;
+}
```

Comment:
this probably shouldn't be all rolled up into one function... I'd just define it separately and differently in `default_upsweep.comp` and `default_downsweep.comp`

Reply:
makes sense